src/Components/Fsi.fs

```diff
@@ -431,10 +430,7 @@ module Fsi =
 
                     failwith "unable to spawn FSI"
 
-            let terminal =
-                match profile.options with
-                | U2.Case1 opts -> window.createTerminal opts
-                | U2.Case2 opts -> window.createTerminal opts
```

Comment:
Ok, so in the general case when we're using erased unions to model VSCode APIs, when 'matching' on those APIs we can't directly match on the case and should instead use methods like probing for specific properties to determine which 'case' we're in?

Reply:
You are right.

You can use type test for native types like `string`.

And for others types, you need to do it as if you were writing JavaScript and so by checking something unique like the constructor instance, a property presence, etc.

You can make it looks nice in F# by using custom active pattern.

Reply:
It's... complicated. As @MangelMaxime says, if you have an erased union with primitives like `U2<string, number>` or actual types `U2<ClassA, ClassB>` runtime type testing will work, but it's not easy to test an structural interface at runtime so for now Fable just emits false in those cases.

We could try to emulate interface type testing by checking the property descriptors of the object, though it still won't work in 100% of cases.